src/engine/IndexScan.cpp

```diff
@@ -64,16 +64,20 @@ IndexScan::IndexScan(QueryExecutionContext* qec, Permutation::Enum permutation,
                      const TripleComponent& o,
                      std::vector<ColumnIndex> additionalColumns,
                      std::vector<Variable> additionalVariables,
-                     Graphs graphsToFilter, PrefilterIndexPair prefilter)
+                     const size_t numVariables, Graphs graphsToFilter,
+                     PrefilterIndexPair prefilter)
     : Operation(qec),
       permutation_(permutation),
       subject_(s),
       predicate_(p),
       object_(o),
       graphsToFilter_(std::move(graphsToFilter)),
       prefilter_(std::move(prefilter)),
+      numVariables_(numVariables),
       additionalColumns_(std::move(additionalColumns)),
-      additionalVariables_(std::move(additionalVariables)) {}
+      additionalVariables_(std::move(additionalVariables)) {
+  std::tie(sizeEstimateIsExact_, sizeEstimate_) = computeSizeEstimate();
```

Comment:
With a prefiltered expression, `sizeEstimateIsExact_` is typically `false`. So you have to change the computeSizeEstimate to reflext prefilters (also the getExactSize function, but we'll have to check, if this needs changes to the interface of the `Permutation` class (in particular: Does it already let you explicitly specify the blocks).